iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractLoadingLimits.java

```diff
@@ -85,6 +92,66 @@ public L setPermanentLimit(double permanentLimit) {
+
+        // verif
+        NetworkImpl network = group.getNetwork();
+        ValidationUtil.checkLoadingLimits(
+                group.getValidable(),
+                getPermanentLimit(),
+                getTemporaryLimits(),
+                network.getMinValidationLevel(),
+                network.getReportNodeContext().getReportNode()
+        );
+
+        TreeMap<Integer, TemporaryLimit> temporaryLimitTreeMap = new TreeMap<>(this.temporaryLimits);
+        // création des repérages par index
+        Map.Entry<Integer, TemporaryLimit> lowerEntry = temporaryLimitTreeMap.lowerEntry(acceptableDuration);
+        Map.Entry<Integer, TemporaryLimit> higherEntry = temporaryLimitTreeMap.higherEntry(acceptableDuration);
+
+        // trouver la limite qu'on veut modifier et sa valeur
+        TemporaryLimit identifiedLimit = getTemporaryLimit(acceptableDuration);
+        if (identifiedLimit == null) {
+            LOGGER.info("No temporary limit found for the given acceptable duration");
+            return (L) this;
+        }
+
+        double oldValue = identifiedLimit.getValue();
+
+        if (isTemporaryLimitVaLueValid(lowerEntry, higherEntry, acceptableDuration, temporaryLimitValue)) {
+            identifiedLimit.setValue(temporaryLimitValue);
+            network.invalidateValidationLevel();
+            group.notifyTemporaryLimitValueUpdate(getLimitType(), oldValue, temporaryLimitValue);
+            LOGGER.info("Temporary limit value changed from {} to {}", oldValue, temporaryLimitValue);
+        } else {
+            LOGGER.info("Temporary limit value couldn't be changed because it is not valid");
+            return (L) this;
+        }
+
+        return (L) this;
+    }
+
+    protected boolean isTemporaryLimitVaLueValid(Map.Entry<Integer, TemporaryLimit> lowerEntry,
+                                               Map.Entry<Integer, TemporaryLimit> higherEntry,
+                                               int acceptableDuration,
+                                               double temporaryLimitValue) {
+
+        if (lowerEntry != null && higherEntry != null) {
+            return lowerEntry.getValue().getAcceptableDuration() > acceptableDuration
+                    && higherEntry.getValue().getAcceptableDuration() < acceptableDuration
+                    && lowerEntry.getValue().getValue() < temporaryLimitValue
+                    && higherEntry.getValue().getValue() > temporaryLimitValue;
+        } else if (lowerEntry == null && higherEntry != null) {
+            return temporaryLimitValue > this.permanentLimit &&
+                    higherEntry.getValue().getValue() > temporaryLimitValue;
+        } else if (lowerEntry != null && higherEntry == null) {
+            return lowerEntry.getValue().getAcceptableDuration() > acceptableDuration
+                    && lowerEntry.getValue().getValue() < temporaryLimitValue;
+        }
+        return false;
```

Comment:
I think you're right, but instead of the default case being `true` , i changed it to the default case being  `return temporaryLimitValue > this.permanentLimit;`, because even if there's only one temporary limit, the permanent one still has to be checked.